tests/evm/test_calldatasize.py

```diff
@@ -0,0 +1,72 @@
+import pytest
+
+from zkevm_specs.evm import (
+    ExecutionState,
+    StepState,
+    Opcode,
+    verify_steps,
+    Tables,
+    RWTableTag,
+    RW,
+    CallContextFieldTag,
+    Bytecode,
+)
+from zkevm_specs.util import rand_fp, RLC, U64
+from zkevm_specs.util.param import N_BYTES_U64
+
+
+TESTING_DATA = (
+    0x00,
+    0x10,
+    0x302010,
+    0xF0FFFFFFFFFFFF0F,
+)
+
+
+@pytest.mark.parametrize("calldatasize", TESTING_DATA)
+def test_calldatasize(calldatasize: U64):
+    randomness = rand_fp()
+
+    bytecode = Bytecode().calldatasize()
+    bytecode_hash = RLC(bytecode.hash(), randomness)
+
+    tables = Tables(
+        block_table=set(),
+        tx_table=set(),
+        bytecode_table=set(bytecode.table_assignments(randomness)),
+        rw_table=set(
+            [
+                (9, RW.Read, RWTableTag.CallContext, 1, CallContextFieldTag.CallDataLength, calldatasize, 0, 0),
+                (10, RW.Write, RWTableTag.Stack, 1, 1023, RLC(calldatasize, randomness, N_BYTES_U64), 0, 0),
+            ]
+        ),
+    )
+
+    verify_steps(
+        randomness=randomness,
+        tables=tables,
+        steps=[
+            StepState(
+                execution_state=ExecutionState.CALLDATASIZE,
+                rw_counter=9,
+                call_id=1,
+                is_root=True,
```

Comment:
@ed255 Should we also test the scenario where this is not the root (`is_root=False`) so that we test `CALLDATASIZE` for internal calls as well?

Reply:
I think that's a very interesting case to test.  Nevertheless I think we'll have to wait for this https://github.com/appliedzkp/zkevm-specs/pull/82 to be merged so that we have `CALL`s in order to achieve a test situation where `is_root=False`.

So I'd say that we can delay this for the future.